bip-xxxx.mediawiki

```diff
@@ -0,0 +1,344 @@
+* If <code>maxadditionalfeecontribution=</code> was not specified, the sender's software should ask an interactive confirmation to the user.
+* <code>maxadditionalfeecontribution=</code> and <code>minfeerate=</code> should be ignored in the [[#spare-change|spare change]] case.
+
+The sender must be careful to only sign the inputs that were present in the original PSBT and nothing else.
+
+Note:
+* The sender should allow the payment output to be modified by the receiver (The receiver may substitute a P2WPKH payment to P2SH payment to increase privacy)
+* The sender must allow the receiver to add outputs.
+* The sender must allow the receiver to not add any input. Useful for the receiver to change the paymout output scriptPubKey type.
+* If no input have been added, the sender's wallet implementation should accept the payjoin proposal, but not mark the transaction as an actual payjoin in the user interface.
+
+Our method of checking the fee allows the receiver and the sender to batch payments in the payjoin transaction.
+It also allows the receiver to pay the fee for batching adding his own outputs.
+
+On top of those check, it is recommended, but not required for the sender to check that:
+* If the sender is making a payjoin with a change (ie, not in the [[#spare-change|spare change]] case), make sure the receiver is paying for any change in the output list.
+
+===Optional parameters===
+
+When the payjoin sender posts the original PSBT to the receiver, he can optionally specify the following HTTP query string parameters:
+
+* <code>v=</code>, the version number of the payjoin protocol that the sender is using. The current version is <code>1</code>.
+
+This can be used in the future so the receiver can reject a payjoin if the sender is using a version which is not supported via an error HTTP 400, <code>version-unsupported</code>.
+If not specified, the receiver will assume the sender is <code>v=1</code>.
+
+If the receiver does not support the version of the sender, they should send an error with the list of supported versions:
+<pre>
+{
+    "errorCode": "version-unsupported",
+    "supported" : [ 2, 3, 4 ],
+    "message": "The version is not supported anymore"
+}
+</pre>
+
+* <code>additionalfeeoutputindex=</code>, the preferred output from which to increase the fee for the added inputs. (default: <code>-1</code>)
+
+If the <code>additionalfeeoutputindex</code> is out of bounds or pointing to the payment ouptut meant for the receiver, the receiver should ignore the parameter.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>maxadditionalfeecontribution=</code>, an integer defining the maximum amount in satoshis that the sender is willing to contribute towards fees for the additional inputs. <code>maxadditionalfeecontribution</code> must be ignored if set to less than zero. (default: -1)
```

Comment:
s/is willing to contribute towards fees for the additional inputs/is willing to contribute towards fees for the additional inputs and outputs/

Reply:
@instagibbs I actually specified that the receiver should not free ride, except in the case of spare change (https://github.com/bitcoin/bips/pull/923/commits/5a337c6fc6104fc31eff0ecbb654e9580af4f907)

My client implementation don't enforce it yet, but IMHO this should be specified.

Reply:
What about

>  amount in satoshis that the sender is willing to contribute towards fees for the additional inputs and spare change's fake output

Reply:
Actually I don't even need to specify this, as in the case of spare change, the sender can't actually pay any fee since there is no change back to him!

Reply:
I'm confused. For example, the receiver could do something like swap out p2pkh in favor of p2wsh deposit address, increasing the transaction's output by a few bytes. Are you saying that the current spec forbids this?